sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -58,6 +58,12 @@ class QueryExecution(
 
   val id: Long = QueryExecution.nextExecutionId
 
+  private var _commandExecutionIdGenerated = false
```

Comment:
Let's try our best to avoid mutable states. It will make the system harder to reason about.